lib/iris/fileformats/pp.py

```diff
@@ -1378,8 +1378,16 @@ def save(self, file_handle):
 
         # Get the actual data content.
         data = self.data
-        if ma.is_masked(data):
-            data = data.filled(fill_value=self.bmdi)
+        if isinstance(data, ma.MaskedArray):
+            mdi = self.bmdi
+            if np.any(data == mdi):
```

Comment:
This is still inconsistent with the proposal at https://github.com/SciTools/iris/issues/2748#issuecomment-324578818. Specifically:

```
warn if any fill-values are present (unmasked points only, in case of masked array)
explain in terms of "this won't read back the same"
```